rep-2003.rst

```diff
@@ -0,0 +1,76 @@
+REP: 2003
+Title:  Sensor Data and Map QoS Settings
+Author: Steve Macenski <stevenmacenski at gmail.com>
+Status: Draft
+Type: Standards Track
+Content-Type: text/x-rst
+Created: 08-Nov-2019
+ROS-Version: Dashing
+Post-History: 08-Nov-2019
+
+Abstract
+========
+
+This REP covers the standards surrounding Quality of Service (QoS) settings for common needs in ROS2.
+This includes sensor drivers and map representations to standardize interfaces of not only message types outlined by REPs such as REP 118, 138, and 145, but also QoS to ensure ROS2 interoperability.
+
+Specification
+=============
+
+Map Quality of Service
+----------------------
+
+Map providers such as map servers and SLAM implementations are expected to provide all maps as over a reliable local-transient topic.
+This is to allow for reliable transmission of map data and provide the map information to any new map clients without delay.
+The depth of the transient-local storage depth is left to the designer, however a single map depth is a reasonable choice for static or globally-updated dynamic map serving applications.
+Map clients are also expected to receive maps over a reliable local-transient topic with variably depth.
+
+This specification does not enforce any particular map representation but rather applies for all shared map representations including but not limited to occupancy grids, feature maps, or dense representations.
+
+Sensor Driver Quality of Service
+--------------------------------
+
+Sensor data provided by a sensor driver from a camera, inertial measurement unit, laser scanner, GPS, depth, or range finder are expected to be provided over a SensorDataQoS quality of service as provided by the implemented ROS2 version API.
```

Comment:
Seeing as this is a generic REP: is this list exhaustive? Or are these just examples?